Ryujinx.HLE/Input/Controller/BaseController.cs

```diff
@@ -0,0 +1,143 @@
+﻿using static Ryujinx.HLE.Input.Hid;
+
+namespace Ryujinx.HLE.Input
+{
+    public abstract class BaseController : IHidDevice
+    {
+        protected ControllerStatus HidControllerType;
+        protected ControllerId     ControllerId;
+
+        private long _currentLayoutOffset;
+        private long _mainLayoutOffset;
+
+        protected long DeviceStateOffset => Offset + 0x4188;
+
+        protected Switch Device { get; }
+
+        public long Offset    { get; private set; }
+        public bool Connected { get; protected set; }
+
+        public ControllerHeader      Header             { get; private set; }
+        public ControllerStateHeader CurrentStateHeader { get; private set; }
+        public ControllerDeviceState DeviceState        { get; private set; }
+        public ControllerLayouts     CurrentLayout      { get; private set; }
+        public ControllerState       LastInputState     { get; set; }
+        public ControllerConnState   ConnectionState    { get; protected set; }
+
+        public BaseController(ControllerStatus controllerType, Switch device)
+        {
+            Device = device;
```

Comment:
Remove empty line and align:
```suggestion
            Device            = device;
            HidControllerType = controllerType;
```